src/hydration/hydrate.ts

```diff
@@ -0,0 +1,143 @@
+import { Float } from 'java.lang';
+import { List } from 'java.util';
+import { Material } from 'org.bukkit';
+import { Block } from 'org.bukkit.block';
+import { Levelled } from 'org.bukkit.block.data';
+import { Player } from 'org.bukkit.entity';
+import { Action } from 'org.bukkit.event.block';
+import {
+  PlayerInteractEvent,
+  PlayerItemConsumeEvent,
+} from 'org.bukkit.event.player';
+import { EquipmentSlot } from 'org.bukkit.inventory';
+
+const Hydration = {
+  MAX: 0.99,
+  MIN: 0.0,
+  HUGE: 0.4,
+  MEDIUM: 0.25,
+  SMALL: 0.05,
+};
+
+// When player drinks (bottles etc)
+registerEvent(PlayerItemConsumeEvent, (event) => {
+  const item = event.getItem();
+
+  const material = item?.getType();
+  const player = event.getPlayer();
+  switch (material) {
+    case Material.POTION:
+      hydrate(player, Hydration.MEDIUM, material);
+      break;
+    case Material.MELON_SLICE:
+      hydrate(player, Hydration.SMALL, material);
+      break;
+  }
+});
+
+// When player drinks from block
+const drinkers: Player[] = [];
```

Comment:
Use ES6 Set for average O(1) (instead of O(n)) for checking if a player is currently drinking.